scrapy/commands/genspider.py

```diff
@@ -69,7 +69,18 @@ def run(self, args, opts):
         try:
             spidercls = self.crawler_process.spider_loader.load(name)
         except KeyError:
-            pass
+            if not self.settings.get('NEWSPIDER_MODULE'):
+                if exists(name + ".py"):
+                    print("File %r already exists in the current directory" % (name + ".py"))
+                    return
+            else:
+                spiders_module = import_module(self.settings['NEWSPIDER_MODULE'])
+                spiders_dir = dirname(spiders_module.__file__)
+                spiders_dir_abs = abspath(spiders_dir)
+                if exists(join(spiders_dir_abs, name + ".py")):
+                    print("Spider with filename %r already exists in directory:" % (name + ".py"))
+                    print("  %s" % spiders_dir)
```

Comment:
:lipstick:  I would simplify both messages to "%r already exists", where %r is the absolute path of the destination file.

Reply:
Seems more concise, will make the change!